eth/tracers/live/statesize.go

```diff
@@ -0,0 +1,711 @@
+	}
+	t.file = file
+	t.writer = csv.NewWriter(file)
+
+	// Write header if file is new (empty)
+	info, err := file.Stat()
+	if err != nil {
+		file.Close()
+		return nil, fmt.Errorf("failed to stat statesize file: %v", err)
+	}
+	if info.Size() == 0 {
+		if err := t.writer.Write(csvHeaders); err != nil {
+			file.Close()
+			return nil, fmt.Errorf("failed to write CSV headers: %v", err)
+		}
+		t.writer.Flush()
+	}
+
+	// Open depth created file for appending
+	depthCreatedFile, err := os.OpenFile(depthCreatedFilePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0o644)
+	if err != nil {
+		file.Close()
+		return nil, fmt.Errorf("failed to open depth created file: %v", err)
+	}
+	t.depthCreatedFile = depthCreatedFile
+	t.depthCreatedWriter = csv.NewWriter(depthCreatedFile)
+
+	// Write header if depth created file is new (empty)
+	depthCreatedInfo, err := depthCreatedFile.Stat()
+	if err != nil {
+		file.Close()
+		depthCreatedFile.Close()
+		return nil, fmt.Errorf("failed to stat depth created file: %v", err)
+	}
+	if depthCreatedInfo.Size() == 0 {
+		if err := t.depthCreatedWriter.Write(depthCSVHeaders()); err != nil {
+			file.Close()
+			depthCreatedFile.Close()
+			return nil, fmt.Errorf("failed to write depth created CSV headers: %v", err)
+		}
+		t.depthCreatedWriter.Flush()
+	}
+
+	// Open depth deleted file for appending
+	depthDeletedFile, err := os.OpenFile(depthDeletedFilePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0o644)
+	if err != nil {
+		file.Close()
+		depthCreatedFile.Close()
+		return nil, fmt.Errorf("failed to open depth deleted file: %v", err)
+	}
+	t.depthDeletedFile = depthDeletedFile
+	t.depthDeletedWriter = csv.NewWriter(depthDeletedFile)
+
+	// Write header if depth deleted file is new (empty)
+	depthDeletedInfo, err := depthDeletedFile.Stat()
+	if err != nil {
+		file.Close()
+		depthCreatedFile.Close()
+		depthDeletedFile.Close()
+		return nil, fmt.Errorf("failed to stat depth deleted file: %v", err)
+	}
+	if depthDeletedInfo.Size() == 0 {
+		if err := t.depthDeletedWriter.Write(depthCSVHeaders()); err != nil {
+			file.Close()
+			depthCreatedFile.Close()
+			depthDeletedFile.Close()
+			return nil, fmt.Errorf("failed to write depth deleted CSV headers: %v", err)
+		}
+		t.depthDeletedWriter.Flush()
+	}
+
+	return &tracing.Hooks{
+		OnStateUpdate: t.onStateUpdate,
+		OnClose:       t.onClose,
+	}, nil
+}
+
+// loadStatsForRoot searches the CSV file for a record with the given root hash
+// and returns its stats. Returns zero stats if not found.
+func (s *stateSizeTracer) loadStatsForRoot(root common.Hash) (stateSizeStats, bool) {
+	file, err := os.Open(s.filePath)
+	if err != nil {
+		return stateSizeStats{}, false
+	}
+	defer file.Close()
+
+	reader := csv.NewReader(bufio.NewReader(file))
+
+	// Read and skip header
+	if _, err := reader.Read(); err != nil {
+		return stateSizeStats{}, false
+	}
+
+	// Search for the root in the CSV file (read backwards would be more efficient,
```

Comment:
Hmm, if the csv file contains the statistics of the entire history (24m entries), the initial bootstrap will be very expensive.